src/renderer/components/dialogs/WebxdcSaveToChatDialog.tsx

```diff
@@ -0,0 +1,150 @@
+  const listFlags = C.DC_GCL_FOR_FORWARDING | C.DC_GCL_NO_SPECIALS
+  const { chatListIds, queryStr, setQueryStr } = useChatList(listFlags)
+  const { isChatLoaded, loadChats, chatCache } = useLogicVirtualChatList(
+    chatListIds,
+    listFlags
+  )
+
+  const onChatClick = async (chatId: number) => {
+    let path = null
+    if (file) {
+      path = await runtime.writeTempFileFromBase64(
+        file.file_name,
+        file.file_content
+      )
+    }
+    await saveToChatAction(chatId, messageText, path)
+    if (path) {
+      await runtime.removeTempFile(path)
+    }
+    onClose()
+  }
+
+  const onSearchChange = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setQueryStr(e.target.value)
+
+  const CHATLISTITEM_CHAT_HEIGHT =
+    Number(useThemeCssVar('--SPECIAL-chatlist-item-chat-height')) || 64
+
+  const noResults = chatListIds.length === 0 && queryStr !== ''
+  return (
+    <DeltaDialogBase isOpen={true} onClose={onClose} fixed>
+      <DeltaDialogHeader
+        onClose={onClose}
+        title={tx('chat_share_with_title')}
+      />
+      <div
+        className={classNames(
+          Classes.DIALOG_BODY,
+          'bp4-dialog-body-no-footer',
+          'mailto-dialog'
+        )}
+      >
+        <Card style={{ padding: '0px' }}>
+          <div className='select-chat-chat-list'>
+            <input
+              className='search-input'
+              onChange={onSearchChange}
+              value={queryStr}
+              placeholder={tx('contacts_enter_name_or_email')}
+              autoFocus
+              spellCheck={false}
+            />
+            {noResults && queryStr && (
+              <PseudoListItemNoSearchResults queryStr={queryStr} />
+            )}
+            <div style={noResults ? { height: '0px' } : {}} className='results'>
+              <AutoSizer>
+                {({ width, height }) => (
+                  <ChatListPart
+                    isRowLoaded={isChatLoaded}
+                    loadMoreRows={loadChats}
+                    rowCount={chatListIds.length}
+                    width={width}
+                    height={height}
+                    itemKey={index => 'key' + chatListIds[index]}
+                    itemHeight={CHATLISTITEM_CHAT_HEIGHT}
+                  >
+                    {({ index, style }) => {
+                      const chatId = chatListIds[index]
+                      return (
+                        <div style={style}>
+                          <ChatListItem
+                            chatListItem={chatCache[chatId] || undefined}
+                            onClick={onChatClick.bind(null, chatId)}
+                          />
+                        </div>
+                      )
+                    }}
+                  </ChatListPart>
+                )}
+              </AutoSizer>
+            </div>
+          </div>
+        </Card>
+      </div>
+    </DeltaDialogBase>
+  )
+}
+
+async function saveToChatAction(
```

Comment:
same here, consider naming it `sendToChatAction`, there are some more inconsistent namings, i stop listing them